src/System.Xml.ReaderWriter/ref/System.Xml.ReaderWriter.cs

```diff
@@ -1401,39 +1382,28 @@ public sealed partial class XmlAtomicValue : System.Xml.XPath.XPathItem, System.
     }
-    [System.Xml.Serialization.XmlRootAttribute("schema", Namespace="http://www.w3.org/2001/XMLSchema")]
     public partial class XmlSchema : System.Xml.Schema.XmlSchemaObject
     {
         public const string InstanceNamespace = "http://www.w3.org/2001/XMLSchema-instance";
         public const string Namespace = "http://www.w3.org/2001/XMLSchema";
         public XmlSchema() { }
-        [System.ComponentModel.DefaultValueAttribute((System.Xml.Schema.XmlSchemaForm)(0))]
-        [System.Xml.Serialization.XmlAttributeAttribute("attributeFormDefault")]
+        [System.ComponentModel.DefaultValueAttribute(System.Xml.Schema.XmlSchemaForm.None)]
         public System.Xml.Schema.XmlSchemaForm AttributeFormDefault { get { throw null; } set { } }
-        [System.Xml.Serialization.XmlIgnoreAttribute]
         public System.Xml.Schema.XmlSchemaObjectTable AttributeGroups { get { throw null; } }
-        [System.Xml.Serialization.XmlIgnoreAttribute]
         public System.Xml.Schema.XmlSchemaObjectTable Attributes { get { throw null; } }
-        [System.ComponentModel.DefaultValueAttribute((System.Xml.Schema.XmlSchemaDerivationMethod)(256))]
-        [System.Xml.Serialization.XmlAttributeAttribute("blockDefault")]
+        [System.ComponentModel.DefaultValueAttribute(System.Xml.Schema.XmlSchemaDerivationMethod.None)]
         public System.Xml.Schema.XmlSchemaDerivationMethod BlockDefault { get { throw null; } set { } }
-        [System.ComponentModel.DefaultValueAttribute((System.Xml.Schema.XmlSchemaForm)(0))]
-        [System.Xml.Serialization.XmlAttributeAttribute("elementFormDefault")]
+        [System.ComponentModel.DefaultValueAttribute(System.Xml.Schema.XmlSchemaForm.None)]
         public System.Xml.Schema.XmlSchemaForm ElementFormDefault { get { throw null; } set { } }
-        [System.Xml.Serialization.XmlIgnoreAttribute]
         public System.Xml.Schema.XmlSchemaObjectTable Elements { get { throw null; } }
-        [System.ComponentModel.DefaultValueAttribute((System.Xml.Schema.XmlSchemaDerivationMethod)(256))]
-        [System.Xml.Serialization.XmlAttributeAttribute("finalDefault")]
+        [System.ComponentModel.DefaultValueAttribute(System.Xml.Schema.XmlSchemaDerivationMethod.None)]
         public System.Xml.Schema.XmlSchemaDerivationMethod FinalDefault { get { throw null; } set { } }
-        [System.Xml.Serialization.XmlIgnoreAttribute]
         public System.Xml.Schema.XmlSchemaObjectTable Groups { get { throw null; } }
-        [System.Xml.Serialization.XmlAttributeAttribute("id", DataType="ID")]
         public string Id { get { throw null; } set { } }
         [System.Xml.Serialization.XmlElementAttribute("import", typeof(System.Xml.Schema.XmlSchemaImport))]
         [System.Xml.Serialization.XmlElementAttribute("include", typeof(System.Xml.Schema.XmlSchemaInclude))]
         [System.Xml.Serialization.XmlElementAttribute("redefine", typeof(System.Xml.Schema.XmlSchemaRedefine))]
```

Comment:
@ericstj, should we be excluding `System.Xml.Serialization.XmlElementAttribute`?

Reply:
Removing any XML Serialization attributes seems a little weird to me... it feels like using a type that had attributes as a property within a custom type and then using `sgen.exe` would produce different answers than if the serialization assembly were dynamically produced at runtime.

(Though I guess that's potentially true anyways, since the impl could have properties that the ref didn't know about... Serialization's hard :smile:)

Reply:
Problem / discussion moved to #35561.

@ahsonkhan For now let's go with "don't change the ref regarding XML Serialization attributes"; so any that got removed in this PR (if the attribute usage has shipped in a release already) should be put back.

Reply:
Agree, lets not remove attributes unless we are 100% they are not needed / implementation only.  XML serialization owners would have to make that call.

Reply:
I don't think we should ever be removing attributes. These, even for reference assemblies, can be consumed by analyzers or external tools for various light-up functionality.

CC. @jaredpar

Reply:
> I don't think we should ever be removing attributes.

If the attributes were added by mistake, it is fine to remove them.

FWIW, I do not believe that we have ever supported generation of serializers from reference assemblies.

Reply:
> If the attributes were added by mistake, it is fine to remove them.

Right. I should clarify that I was talking about a general: "Always remove this attribute" scenario. Removed attributes should probably be a case-by-case basis with explicit reasoning on why. 

Everywhere we differ from what the C# compiler would generate it is `refout` feature is just a potential bug waiting to crop up down the road.